tests/checkers/unittest_non_ascii_name.py

```diff
@@ -0,0 +1,312 @@
+            ),
+            pytest.param(
+                """
+                class FooBar:
+                    łol = "test" #@
+                """,
+                "Attribute",
+                id="class_attribute",
+            ),
+            pytest.param(
+                """
+                łol = "test" #@
+                """,
+                "Variable",
+                id="global_assign",
+            ),
+            pytest.param(
+                """
+                def foobar():
+                    łol="test"  #@
+                """,
+                "Variable",
+                id="function_variable",
+            ),
+            pytest.param(
+                """
+                for łol in os.listdir("."):  #@
+                    ...
+                """,
+                "Variable",
+                id="for_loop_variable",
+            ),
+            pytest.param(
+                """
+                [łoł
+                    for łol in os.listdir(".")  #@
+                ]
+                """,
+                "Variable",
+                id="inline_for_loop_variable",
+            ),
+        ],
+    )
+    def test_assignname(
+        self,
+        code: str,
+        assign_type: str,
+    ):
+        """Variables defined no matter where, should be checked for non ascii"""
+        assign_node = astroid.extract_node(code)
+
+        if not isinstance(assign_node, nodes.AssignName):
+            # For some elements we can't directly extract the assign
+            # node, so we have to manually look in the children for it
+            for child in assign_node.get_children():
+                if isinstance(child, nodes.AssignName):
+                    assign_node = child
+                    break
+
+        # Just to make sure we found the correct node
+        assert isinstance(assign_node, nodes.AssignName)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=assign_node,
+                args=(assign_type, "łol"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_assignname(assign_node)
+
+    def test_class_definition(self):
+        node = astroid.extract_node(
+            """
+            class FooBär: #@
+                ...
+            """
+        )
+        assert isinstance(node, nodes.ClassDef)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=node,
+                args=("Class", "FooBär"),
+                confidence=pylint.interfaces.HIGH,
+                line=2,
+                col_offset=0,
+                # pylint: disable-next=fixme
+                # TODO: Maybe we have to select something different
+                #       here, as line 3 is not that thing we expect
+                end_line=3,
```

Comment:
I wonder what's happening here.. Is the Ellipsis considered part of the `ClassDef` node?

Reply:
Yes I think so. Probably the complete class (defintion) is considered.
Maybe we can define the end_line manually to line, as a Class Defintion (at least the part containing the name) shouldn't exceed one line? Is there an easy way to do this?

Reply:
This is something we're thinking about in #5466.

We'll likely change it so that `add_message` detects class and function nodes and handles them a bit better. For now this is fine.

(Thinking about this, this is another example of where the unittests do not represent actual pylint. The tests use a different `add_message` function which is defined independently of the actual function. So, if we change anything to `PyLinter.add_message` this doesn't get reflected in the unittests. Imo, this is a major design flaw of the unittests that has probably been here for years, but also one of the reasons why I often ask people to write functional tests.)

Reply:
Removed test, as seems to be  more of an integration issues = functional test